main.py

```diff
@@ -0,0 +1,59 @@
+import sys
+
+import streamlit as st
+import torch
+from diffusers import StableDiffusionPipeline
+import requests
+from bs4 import BeautifulSoup
+from googletrans import Translator
+def check_url(url):
+    r = requests.head(url)
+    return r.status_code == 200
+
+def get_text_from_site(url):
+    try:
+        response = requests.get(url)
+        bs = BeautifulSoup(response.text, 'html.parser')
+        part_of_text = bs.find('div', 'c-detail text').text
+        index = part_of_text.lower().find("приметы")
+        i = part_of_text.find(":", index)
+        result_sentence = part_of_text[i+1:part_of_text.find("\n", i+1)].replace(".", "")
+        translator = Translator()
+        translation = translator.translate(result_sentence, src='ru', dest='en')
+        result_sentence = translation.text
+    except:
+        result_sentence = ""
+    return result_sentence
+
+def generate_arcane(text):
+    model_id = "nitrosocke/Arcane-Diffusion"
+    if torch.cuda.is_available():
+        pipe = StableDiffusionPipeline.from_pretrained(model_id, torch_dtype=torch.float16)
+        pipe = pipe.to("cuda")
+    else:
+        pipe = StableDiffusionPipeline.from_pretrained(model_id)
+        pipe = pipe.to("cpu")
+    pipe.safety_checker = lambda images, clip_input: (images, False)
+    prompt = "arcane style, " + text
+    images = pipe(prompt).images
```

Comment:
There are some parameters in the pipeline that can improve or speed up generation. You can also generate multiple images to see all possible suspect visualizations. See this guide https://huggingface.co/blog/stable_diffusion.

I think it would be great to add multiple-image support. And you can hide generation parameters under the `Advance` tab, so anyone can play with them in case they know what it is.